tests/test_sap.py

```diff
@@ -269,3 +269,211 @@ def test_generate_report_success():
+                    "postal code": "94109",
+                    "country": "US",
+                },
+            },
+            "funds": {
+                "420800-1614906": {
+                    "amount": 150,
+                    "G/L account": "420800",
+                    "cost object": "1614906",
+                },
+            },
+        },
+        {
+            "date": datetime(2021, 5, 11),
+            "id": "5470276540006761",
+            "number": "883299",
+            "type": "monograph",
+            "payment method": "ACCOUNTINGDEPARTMENT",
+            "total amount": 1067.04,
+            "currency": "USD",
+            "vendor": {
+                "name": "GOBI Library Solutions from EBSCO",
+                "code": "YBPE-M",
+                "address": {
+                    "lines": [
+                        "P.O. Box 277991",
+                    ],
+                    "city": "Atlanta",
+                    "state or province": "GA",
+                    "postal code": "30384-7991",
+                    "country": "US",
+                },
+            },
+            "funds": {
+                "420800-1614906": {
+                    "amount": 608,
+                    "G/L account": "420800",
+                    "cost object": "1614906",
+                },
+                "420800-2480602": {
+                    "amount": 148.50,
+                    "G/L account": "420800",
+                    "cost object": "2480602",
+                },
+                "1420800-3164500": {
+                    "amount": 235.54,
+                    "G/L account": "420800",
+                    "cost object": "3164500",
+                },
+                "420800-3203000": {
+                    "amount": 75,
+                    "G/L account": "420800",
+                    "cost object": "3203000",
+                },
+            },
+        },
+        {
+            "date": datetime(2021, 5, 12),
+            "id": "5470276540006761",
+            "number": "247623",
+            "type": "monograph",
+            "payment method": "ACCOUNTINGDEPARTMENT",
+            "total amount": 150,
+            "currency": "USD",
+            "vendor": {
+                "name": "one address line",
+                "code": "KANOPY-M",
+                "address": {
+                    "lines": [
+                        "123 some street",
+                    ],
+                    "city": "San Francisco",
+                    "state or province": "CA",
+                    "postal code": "94109",
+                    "country": "US",
+                },
+            },
+            "funds": {
+                "420800-1614906": {
+                    "amount": 150,
+                    "G/L account": "420800",
+                    "cost object": "1614906",
+                },
+            },
+        },
+    ]
+    report = sap.generate_sap_data(today, invoices)
+    assert report == (
+        "B\
```

Comment:
yes, I intentionally made it one line per field so that it would be more readable and easier to confirm that the fields were the right length. I'll add a comment saying as much.